presto-main/src/main/java/com/facebook/presto/server/HttpRequestSessionContext.java

```diff
@@ -432,6 +438,32 @@ public boolean supportClientTransaction()
         return traceToken;
     }
 
+    @Override
+    public Optional<Tracer> getTracer()
+    {
+        return tracer;
+    }
+
+    @Override
+    public void setTracerIfEnabled(Tracer tracer)
+    {
+        if (isTracingEnabled()) {
+            this.tracer = Optional.of(requireNonNull(tracer, "tracer is null"));
+        }
+        else {
+            this.tracer = Optional.of(NoopTracerProvider.NOOP_TRACER);
+        }
+    }
+
+    private boolean isTracingEnabled()
+    {
+        if (this.systemProperties.getOrDefault(SystemSessionProperties.ENABLE_DISTRIBUTED_TRACING, "")
+                .equalsIgnoreCase("true")) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
add a static helper to SystemSessionProperties. There are many examples in that class

Reply:
Session is not created yet when tracing starts. But SessionContext (HttpRequestSessionContext) is created at the time. That is why I cannot find a proper way to get the enable_distributed_tracing session property at starting point from Session directly (Hence not able to leverage SystemSessionProperties). I wonder if there are some better ways?